tools/benchmarking/benchmark.py

```diff
@@ -0,0 +1,209 @@
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Dict, List, Union, cast
+
+import pandas as pd
+from omegaconf import DictConfig, ListConfig, OmegaConf
+from pytorch_lightning import Trainer, seed_everything
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.callbacks.model_checkpoint import ModelCheckpoint
+
+from anomalib.config import get_configurable_parameters, update_input_size_config
+from anomalib.core.callbacks import get_callbacks
+from anomalib.core.model.anomaly_module import AnomalyModule
+from anomalib.data import get_datamodule
+from anomalib.models import get_model
+from anomalib.utils.optimize import export_convert
+from anomalib.utils.sweep import get_run_config, set_in_nested_config
+from anomalib.utils.sweep.helpers.inference import (
+    get_meta_data,
+    get_openvino_throughput,
+    get_torch_throughput,
+)
+
+
+def convert_to_openvino(model: AnomalyModule, export_path: Union[Path, str], input_size: List[int]):
+    """Convert the trained model to OpenVINO."""
+    export_path = export_path if isinstance(export_path, Path) else Path(export_path)
+    onnx_path = export_path / "model.onnx"
+    export_convert(model, input_size, onnx_path, export_path)
+
+
+def update_callbacks(config: Union[DictConfig, ListConfig]) -> List[Callback]:
+    """Disable callbacks in config.
+
+    Args:
+        config (Union[DictConfig, ListConfig]): Model config loaded from anomalib
+    """
+    config.project.log_images_to = []  # disable visualizer callback
+
+    # disable openvino optimization
+    if "optimization" in config.keys():
+        config.pop("optimization")
+
+    # disable load model callback
+    if "weight_file" in config.model.keys():
+        config.model.pop("weight_file")
+
+    # disable save_to_csv. Metrics will be captured from stdout
+    if "save_to_csv" in config.project.keys():
+        config.project.pop("save_to_csv")
+
+    callbacks = get_callbacks(config)
+
+    # remove ModelCheckpoint callback
+    for index, callback in enumerate(callbacks):
+        if isinstance(callback, ModelCheckpoint):
+            callbacks.pop(index)
+            break
+
+    return callbacks
+
+
+def get_single_model_metrics(model_config: Union[DictConfig, ListConfig], openvino_metrics: bool = False) -> Dict:
+    """Collects metrics for `model_name` and returns a dict of results.
+
+    Args:
+        model_config (DictConfig, ListConfig): Configuration for run
+        openvino_metrics (bool): If True, converts the model to OpenVINO format and gathers inference metrics.
+
+    Returns:
+        Dict: Collection of all the metrics such as time taken, throughput and performance scores.
+    """
+
+    with TemporaryDirectory() as project_path:
+        model_config.project.path = project_path
+        datamodule = get_datamodule(model_config)
+        model = get_model(model_config)
+
+        callbacks = update_callbacks(model_config)
```

Comment:
Instead of removing the unnecessary callbacks with `update_callbacks`, could we just create a new list of callbacks and pass that to the trainer? Which callbacks do we need during benchmarking anyway?